c7n/resources/eks.py

```diff
@@ -1,16 +1,87 @@
 # Copyright The Cloud Custodian Authors.
 # SPDX-License-Identifier: Apache-2.0
+from botocore.exceptions import ClientError
+
 from c7n.actions import Action
 from c7n.filters.vpc import SecurityGroupFilter, SubnetFilter, VpcFilter
 from c7n.manager import resources
-from c7n import tags
-from c7n.query import QueryResourceManager, TypeInfo, DescribeSource
-from c7n.utils import local_session, type_schema
+from c7n import tags, query
+from c7n.query import QueryResourceManager, ChildResourceManager, ChildDescribeSource, TypeInfo, DescribeSource
+from c7n.utils import local_session, type_schema, get_retry
 from botocore.waiter import WaiterModel, create_waiter_with_client
 from .aws import shape_validate
 from .ecs import ContainerConfigSource
 
 
+@query.sources.register('describe-eks-nodegroup')
+class NodeGroupDescribeSource(ChildDescribeSource):
+
+    def get_query(self):
+        query = super(NodeGroupDescribeSource, self).get_query()
+        query.capture_parent_id = True
+        return query
+
+    def augment(self, resources):
+        results = []
+        client = local_session(self.manager.session_factory).client('eks')
+        for clusterName, nodegroupName in resources:
+            nodegroup = client.describe_nodegroup(
+                clusterName=clusterName,
+                nodegroupName=nodegroupName)['nodegroup']
+            if 'tags' in nodegroup:
+                nodegroup['Tags'] = [{'Key': k, 'Value': v} for k, v in nodegroup['tags'].items()]
+            results.append(nodegroup)
+        return results
+
+
+@resources.register('eks-nodegroup')
+class NodeGroup(ChildResourceManager):
+
+    chunk_size = 10
+
+    class resource_type(TypeInfo):
+
+        service = 'eks'
+        arn = 'arn'
+        arn_type = 'nodegroup'
+        name = id = 'nodegroup-name'
+        enum_spec = ('list_nodegroups', 'nodegroups', None)
+        detail_spec = ('describe_nodegroup', 'nodegroupName', 'clusterName', None)
+        parent_spec = ('eks', 'clusterName', None)
+        permissions_enum = ('eks:DescribeNodegroup',)
+        date = 'createdAt'
+
+    source_mapping = {
+        'describe-child': NodeGroupDescribeSource,
+        'describe': NodeGroupDescribeSource,
+    }
+
+
+@NodeGroup.action_registry.register('delete')
+class DeleteNodeGroup(Action):
+    """Delete node group(s)."""
+
+    schema = type_schema('delete')
+    permissions = ('eks:DeleteNodegroup',)
+
+    def process(self, resources):
+        client = local_session(self.manager.session_factory).client('eks')
+        retry = get_retry(('Throttling',))
+        waiters = []
```

Comment:
generally our actions don't wait for async operations, was there any particular reason for it here? this might be more appropriate for the test validation code

Reply:
I had reused logic of EKS deletion (EKS node groups are deleted because EKS Cluster can't be deleted if node groups exist).

Thanks for info.
Updated.